src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  let year = formatAgan[0];
+  let month = formatAgan[1];
+  let day = formatAgan[2];
```

Comment:
The variable `formatAgan` is initialized as an empty array, and the immediately following lines try to destructure it into `year`, `month`, and `day`. This will not work correctly because `formatAgan` is empty at this point, and destructuring will result in `undefined` values for `year`, `month`, and `day`. Instead, you should be assigning these variables after you have populated the `newDate` array with the split date parts.